ocs_ci/ocs/cluster_load.py

```diff
@@ -18,263 +19,257 @@ class ClusterLoad:
-        self, pvc_factory, pod_factory, target_percentage=0.3, cluster_limit=None
-    ):
+    def reach_cluster_load_percentage_in_throughput(self, ):
         """
-        Reach the cluster throughput limit and then drop to the requested target percentage.
+        Reach the cluster throughput limit and then drop to the given target percentage.
         The number of pods needed for the desired target percentage is determined by
         creating pods one by one, while examining if the cluster throughput is increased
         by more than 10%. When it doesn't increased by more than 10% anymore after
         the new pod started running IO, it means that the cluster throughput limit is
         reached. Then, the function deletes the pods that are not needed as they
-        are the difference between the limit (100%) and the target percentage
-        (the default target percentage is 30%). This leaves the number of pods needed
-        running IO for cluster throughput to be around the desired percentage.
-
-        Args:
-            pvc_factory (function): A call to pvc_factory function
-            pod_factory (function): A call to pod_factory function
-            target_percentage (float): The percentage of cluster load that is required.
-                The value should be greater than 0 and smaller than 1
-            cluster_limit (float): The cluster pre-known throughput limit in Mb/s.
-                If passed, the function will calculate the target throughput based on it
-                multiplied by 'target_percentage'
-
-        Returns:
-            tuple: The cluster limit in Mb/s (float) and the current throughput
-                in Mb/s (float)
+        are the difference between the limit (100%) and the target percentage.
+        This leaves the number of pods needed running IO for cluster throughput to
+        be around the desired percentage.
 
         """
+        if not 0.1 < self.target_percentage < 0.95:
+            self.logger.warning(
+                f"The target percentage is {self.target_percentage * 100}% which is not "
+                f"within the accepted range. Therefore, IO will not be started"
+            )
+            return
+
         pvc_objs = list()
         pod_objs = list()
         limit_reached = False
 
-        # FIO params:
-        # 'runtime' is set with a large value of seconds to
-        # make sure that the pods are running
+        # IO params:
+        storage_type = 'block'
         io_run_time = 100**3
         rate = '200M'
         bs = '128K'
+        rw_ratio = 25
+        pvc_size = int(get_osd_pods_memory_sum() * 0.5)
+        io_file_size = f"{pvc_size - 1}G"
 
-        if 0.1 < target_percentage > 0.95:
-            self.logger.warning(
-                f"The target percentage is {target_percentage * 100}% which is not "
-                f"within the accepted range. Therefore, IO will not be started"
+        def create_pvc_and_pod():
```

Comment:
This is a suggestion only. From my experience creating pods will tend to stick to 2 nodes out of 3 for most of them. I think for better throughput results divide them manually between nodes.

Reply:
That's a very good input!
I will need to check if indeed pods are scheduled only on 2 nodes out of 3, but pinning a pod to a specific node could be a bit problematic as sometimes a node cannot schedule new pods due to lack of resources